src/util/fork_utils.js

```diff
@@ -38,11 +38,22 @@ function start_workers(metrics_port, count = 0) {
         // We don't want to leave the process with a partial set of workers,
         // so if any worker exits, we will print an error message in the logs and start a new one.
         cluster.on('exit', (worker, code, signal) => {
-            console.warn('WORKER exit', { id: worker.id, pid: worker.process.pid, code, signal }, 'starting a new one.');
+            console.warn('WORKER exit', { id: worker.id, pid: worker.process.pid, code, signal });
             new NoobaaEvent(NoobaaEvent.FORK_EXIT).create_event(undefined, { id: worker.id, pid: worker.process.pid,
                 code: code, signal: signal}, undefined);
+            // SIGABRT is used for abnormal program termination - https://www.qnx.com/developers/docs/6.4.1/dinkum_en/ecpp/stdlib.html
+            // we saw it happen both in case of no memory and on too much processes started
+            // in these case we won't restart the workers and will kill the main process
+            // for testing (linux only) limit memory(virt): ulimit -v xxx(kb) 
+            // limit user processes: ulimit -u xxx
+            if (signal === 'SIGABRT') {
+                console.error('EXIT ON WORKER ERROR');
+                new NoobaaEvent(NoobaaEvent.ENDPOINT_CRASHED).create_event(undefined, { id: worker.id, pid: worker.process.pid,
```

Comment:
Without any message that says that we are restarting the main process, someone might think that this process crash happened unexpectedly. 
Maybe add to the arguments a message that describes that this is a crash by us or a new RESTART event, what ever works for you?

Reply:
added the error